### PR TITLE
✨ Add woodpecker as known CI

### LINKED ISSUE #4210: Feature: Recognize Woodpecker-CI as a well-known CI
Add support for [Woodpecker CI](https://woodpecker-ci.org) as it is a nice slim selfhosted CI/CD tool ...

### PR DESCRIPTION
**Pull request by 6543**
#### What kind of change does this PR introduce?

It addresses #4210

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?

It does not count tests run by Woodpecker
e.g. `CI-Tests` score of woodpecker's own main repo is 8/10

#### What is the new behavior (if this is a feature change)?**

it detects woodpecker and score the tests acordingly
e.g. `CI-Tests` score of woodpecker's own main repo is 10/10

`scorecard --repo=https://github.com/woodpecker-ci/woodpecker `:
image

#### Which issue(s) this PR fixes

Fixes #4210

#### Does this PR introduce a user-facing change?

no

### CODE REVIEW

File: probes/testsRunInCI/impl.go [hunk 1]
```diff
@@ -165,7 +165,7 @@ func isTest(s string) bool {
 	for _, pattern := range []string{
 		"appveyor", "buildkite", "circleci", "e2e", "github-actions", "jenkins",
 		"mergeable", "packit-as-a-service", "semaphoreci", "test", "travis-ci",
-		"flutter-dashboard", "Cirrus CI", "azure-pipelines",
+		"flutter-dashboard", "Cirrus CI", "azure-pipelines", "woodpecker",
```
Comment (spencerschrock) [inline probes/testsRunInCI/impl.go:168]: Can we make this more specific to cut down on false positives? Based on the test data, does would `"ci/woodpecker"` work here?

**Review by spencerschrock — APPROVED** [review]
Overall looks good thanks! One quick question

File: probes/testsRunInCI/impl.go (hunk 1 above)
Comment (6543) [reply probes/testsRunInCI/impl.go:168]: Well yes as this is the default string:

https://github.com/woodpecker-ci/woodpecker/blob/b1cf44ae0f80ca18dff3ab67b61cda59107b5a39/cmd/server/flags.go#L231

Comment (6543) [inline probes/testsRunInCI/impl.go:168]:
```suggestion
		"flutter-dashboard", "Cirrus CI", "azure-pipelines", "ci/woodpecker",
```

Comment (6543) [reply probes/testsRunInCI/impl.go:168]: @spencerschrock should i just append a commit or squash merge?

**Review by spencerschrock — APPROVED** [review]
LGTM thanks! Can you update your branch?

Either via rebase or merge is fine, we squash to merge in the end.

**Comment by 6543** [comment]
Done

**Pull request merged**